### PR TITLE
Cleaning up RestResource base class.

### PR DESCRIPTION
**Pull request by dennisoelkers**
* Removing helper methods to the class they are used in
* Replacing restrictToMaster() helper method with @RestrictToMaster annotation

### CODE REVIEW

File: graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFilter.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import javax.ws.rs.container.ContainerRequestFilter;
+import java.io.IOException;
+
+@Priority(Priorities.AUTHORIZATION)
+public class RestrictToMasterFilter implements ContainerRequestFilter {
+    private static final Logger LOG = LoggerFactory.getLogger(RestrictToMasterFilter.class);
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        LOG.warn("Rejected request that is only allowed against master nodes. Returning HTTP 403.");
```
Comment (joschi) [inline graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFilter.java]: I would log this on INFO level (so that most people still see it but don't take it as a kind of error).

File: graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFeature.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class RestrictToMasterFeature implements DynamicFeature {
+    private final ServerStatus serverStatus;
+
+    @Inject
+    public RestrictToMasterFeature(ServerStatus serverStatus) {
+        this.serverStatus = serverStatus;
+    }
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
```
Comment (joschi) [inline graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFeature.java]: Not completely sure about that, but wouldn't this basically add `RestrictToMasterFilter` to every resource class which contains one or more `@RestrictToMaster` annotations so that no resource method of that resource class could be called on non-master Graylog instances whatsoever?

File: graylog2-shared/src/main/java/org/graylog2/shared/rest/resources/RestResource.java [hunk 3]
```diff
@@ -59,9 +59,6 @@
     protected UserService userService;
 
     @Inject
-    protected ServerStatus serverStatus;
```
Comment (joschi) [inline graylog2-shared/src/main/java/org/graylog2/shared/rest/resources/RestResource.java]: Please run "Optimize imports" on this class to remove unneeded imports.

File: graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFilter.java (hunk 1 above)
Comment (dennisoelkers) [inline graylog2-shared/src/main/java/org/graylog2/shared/security/RestrictToMasterFilter.java]: Hm, I am not sure about that. Actually this should not happen™ as the client should not send a request doing master-only actions to a non-master node, so this is kind of an error.

File: graylog2-shared/src/main/java/org/graylog2/shared/rest/resources/RestResource.java (hunk 3 above)
Comment (dennisoelkers) [inline graylog2-shared/src/main/java/org/graylog2/shared/rest/resources/RestResource.java]: Check.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/dashboards/DashboardsResource.java [hunk 4]
```diff
@@ -128,8 +128,6 @@ public Response create(@ApiParam(name = "JSON body", required = true) CreateDash
             @ApiResponse(code = 403, message = "Request must be performed against master node.")
     })
     public DashboardList list() {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/dashboards/DashboardsResource.java]: I think there should be a `@RestrictToMaster` annotation before `list()`, no?

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/MessagesResource.java [hunk 5]
```diff
@@ -77,4 +77,8 @@ public MessagesResource(SystemMessageService systemMessageService) {
                 "messages", messages,
                 "total", systemMessageService.totalCount());
     }
+
+    private int page(int page) {
+        return Math.max(0, page - 1);
+    }
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/MessagesResource.java]: This is an unrelated change?

Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/MessagesResource.java]: No, it is related to the general cleanup which involved moving helper methods to the resources where they are used in.
